shared/extension/Configuration.hpp

```diff
@@ -217,42 +223,52 @@ namespace extension {
 // NUClear configuration extension
 namespace NUClear::dsl {
     namespace operation {
+
+        /// @brief utility function to fetch the first command line argument, used to get the binary name
+        /// @returns If the first command line argument exists, it is returned. Otherwise, the empty string is returned
+        [[nodiscard]] inline std::string getFirstCommandLineArg() {
+            std::shared_ptr<const message::CommandLineArguments> args =
+                store::DataStore<message::CommandLineArguments>::get();
+            if (!args->empty()) {
+                std::vector<char> data(args->at(0).cbegin(), args->at(0).cend());
+                data.push_back('\0');
+                return std::string(data.data());
+            }
+            return std::string{};  // Empty string
+        }
```

Comment:
Can this go in `shared/utility/support/`, it seems a much better home?

Reply:
It's 100% an extension not a utiltiy

Reply:
How does `getFirstCOmmandLineArg()` extend nuclear?

Also can this be snake_case?

Reply:
It's mainly that I don't see this as a utility, but part of the configuration. The fact that it's screwing with the internals of NUClear just makes it feel wrong to consider it anything that's not super tightly coupled to NUClear itself

Reply:
It's also not used anywhere except this file, which goes against our (albeit not strict) policy of making things utilities if they are used by more than one thing

Reply:
Regarding snake case, I was trying to be consistent with the other functions, but now I see that it's only mergeYAML which isn't snake case, in terms of functions in this file. I'll change it too